packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -693,9 +693,6 @@ export class AppSyncModelVisitor<
               isList: false,
               isNullable: field.isNullable,
             });
-          } else if (connectionInfo.targetName !== 'id') {
-            // Need to remove the field that is targetName
-            removeFieldFromModel(model, connectionInfo.targetName);
```

Comment:
Don't we still need to remove if it's not 'id'?

Reply:
The reason I'm not removing anymore here is because belongsTo will never add fields anymore because of my change, and this breaks logic because hasMany will add its connecting field on the connected model. So when hasMany and belongsTo both reference that connected field, this code would snip it out even though it's necessary. This behavior prevents belongsTo from working correctly with DataStore

Reply:
I think V2 transformer uses different approach and there is no need to remove non-id target fields